src/Fantomas/CodePrinter.fs

```diff
@@ -1322,13 +1322,64 @@ and genExpr astContext synExpr =
             else
                 sepSpace ctx
 
-        atCurrentColumn
-            (genExpr astContext e
-             +> colPre sepSpace sepSpace es (fun e ->
-                    onlyIf (isCompExpr e) (sepSpace +> sepOpenSFixed +> sepSpace)
+        // This checks if there is an application with multiple parameters which would fit in one line
+        // but at least one of the parameters which is not the last parameter has a line comment 
+        // In that case we have to split the arguments over multiple lines to preserve the line comments
+        // Example:
+        //
+        //     someFunc a b c // comment
+        //
+        // This can stay as is because the last argument has the line comment
+        // However:
+        //
+        //     someFunc
+        //         a 
+        //         b // comment
+        //         c
+        //
+        // This has to be split over multiple lines to preserve the line comment
+        let parametersMustBeNewlinedBecauseOfTriviaContent es ctx =
```

Comment:
For what unit test did you introduce this? I would assume that a comment in the expression will force `expressionFitsOnRestOfLine` to take the fallback route so I'm not quite sure this is necessary here.